cloudsmith/data_source_package_download.go

```diff
@@ -0,0 +1,150 @@
+package cloudsmith
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// Hit the API with function PackagesListExecute with the following query: name and version (defaults to latest unless otherwise specified)
```

Comment:
convention for function level comments in Go is to start with the function name e.g. `// getPackageURL does the thing ...`